processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/holder/UnsafeSortTempFileChunkHolder.java

```diff
@@ -105,18 +106,24 @@
    * Constructor to initialize
    */
   public UnsafeSortTempFileChunkHolder(File tempFile, SortParameters parameters,
-      boolean convertNoSortFields) {
+      boolean convertNoSortFields, TableFieldStat tableFieldStat) {
     // set temp file
     this.tempFile = tempFile;
     this.readBufferSize = parameters.getBufferSize();
     this.compressorName = parameters.getSortTempCompressorName();
-    this.tableFieldStat = new TableFieldStat(parameters);
+    this.tableFieldStat = tableFieldStat;
     this.sortStepRowHandler = new SortStepRowHandler(tableFieldStat);
     this.executorService = Executors.newFixedThreadPool(1);
-    comparator = new IntermediateSortTempRowComparator(parameters.getNoDictionarySortColumn(),
-        parameters.getNoDictDataType());
     this.convertNoSortFields = convertNoSortFields;
     this.sortTempRowUpdater = tableFieldStat.getSortTempRowUpdater();
+    if (!this.convertNoSortFields) {
```

Comment:
Other two places, IntermediateSortTempRowComparator is directly replaced with FileMergeSortComparator.

Need to support old and new comparator for other two places also ?
If no need to handle other two places implies that  new comparator handles it, so here also no need to handle ?

Reply:
new comparator comes into picture when the carbonrow is updated to schema order, so the sort column went to last, so only in case of final merging only this comparator is used, else it should go to current comparator as before final merge the carbon row has all sort columns at beginning only .(This is done with the help of  method updateDataFieldsBasedOnSortColumns)

Reply:
ok